### PR TITLE
flottbot: init at 0.13.0

### PR DESCRIPTION
**Pull request by bryanhonof**
## Description of changes

cc: @Janik-Haag 

Initialize [flottbot](https://github.com/target/flottbot) at version 0.13.0.
Continuation of https://github.com/NixOS/nixpkgs/pull/229444

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fl/flottbot/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}:
+buildGoModule rec {
+  pname = "flottbot";
+  version = "0.13.0";
+
+  src = fetchFromGitHub {
+    owner = "target";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-ldWE5QcLHyIqap5Qe6OTTIJZ1sshI+CVoJoRUxWHfxM=";
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:15]:
```suggestion
    hash = "sha256-ldWE5QcLHyIqap5Qe6OTTIJZ1sshI+CVoJoRUxWHfxM=";
```

File: pkgs/by-name/fl/flottbot/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "A chatbot framework written in Go";
+    homepage = "https://github.com/target/flottbot";
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.bryanhonof ];
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:39]:
```suggestion
    maintainers = with maintainers; [ bryanhonof ];
```

File: pkgs/by-name/fl/flottbot/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  vendorHash = "sha256-XRcTp3ZnoPupzI1kjoM4oF5+VlNJFV0Bu+WAwfRWl7g=";
+
+  subPackages = [ "cmd/flottbot" ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:35]:
```suggestion
  meta = with lib; {
```

File: pkgs/by-name/fl/flottbot/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    description = "A chatbot framework written in Go";
+    homepage = "https://github.com/target/flottbot";
+    license = lib.licenses.asl20;
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:38]:
```suggestion
    license = licenses.asl20;
```

File: pkgs/by-name/fl/flottbot/package.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    maintainers = [ lib.maintainers.bryanhonof ];
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:40]:
```suggestion
    sourceProvenance = [ sourceTypes.fromSource ];
```

File: pkgs/by-name/fl/flottbot/package.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    mainProgram = "flottbot";
+    platforms = lib.platforms.unix;
```
Comment (mfrw) [inline pkgs/by-name/fl/flottbot/package.nix:42]:
```suggestion
    platforms = platforms.unix;
```

**Review by mfrw — COMMENTED** [review]
A few comments with respect to coherence with other packages :)

File: pkgs/by-name/fl/flottbot/package.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+, nix-update-script
+, substituteAll
+}:
+buildGoModule rec {
+  pname = "flottbot";
+  version = "0.13.0";
+
+  src = fetchFromGitHub {
+    owner = "target";
+    repo = pname;
```
Comment (kirillrdy) [inline pkgs/by-name/fl/flottbot/package.nix:13]: usage of pname is considered an anti pattern https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/fl/flottbot/version.patch [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+ 
+ import (
+ 	"fmt"
+-	"runtime/debug"
+-
+-	"github.com/Masterminds/semver/v3"
+ )
+ 
+ // Version supplies the semantic version.
+@@ -14,20 +11,8 @@ var Version string
```
Comment (kirillrdy) [inline pkgs/by-name/fl/flottbot/version.patch:15]: how about replacing this patch with 

`ldflags=" -X github.com/flattbot/version.Version=${version}`  ?

File: pkgs/by-name/fl/flottbot/package.nix (hunk 7 above)
Comment (bryanhonof) [reply pkgs/by-name/fl/flottbot/package.nix]: Didn't know about this, thanks!

File: pkgs/by-name/fl/flottbot/version.patch (hunk 8 above)
Comment (kirillrdy) [reply pkgs/by-name/fl/flottbot/version.patch:15]: I don't have strong opinion, I am just offering options.

patch is more correct, ldflags is less code :-)

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 290084` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>flottbot</li>
  </ul>
</details>

Tested locally:
```bash
flottbot --version
Version : 0.13.0
Git Hash: 0.13.0
```

**Pull request merged**